src/main/MQ2Commands.cpp

```diff
@@ -4958,151 +4959,126 @@ void SetLootFilterCmd(SPAWNINFO* pChar, char* szLine)
-			GetMaybeQuotedArg(itemFilterArg, MAX_STRING, szLine, 2);
-
-			if (itemFilterArg == nullptr)
-			{
-				WriteChatf("Invalid Item Filter Argument.");
-			}
-
-			// Split out each item in the item filter argument
-			std::vector<std::string> itemFilterStrSplit;
-			std::stringstream ss(itemFilterArg);
-			std::string item;
-			while (getline(ss, item, '^')) {
-				itemFilterStrSplit.push_back(item);
-			}
-
-			// Did not find 3 items in the item filter argument
-			if (itemFilterStrSplit.size() != 3)
-			{
-				WriteChatf("Invalid Item Filter Argument. Check Formatting.");
-				return;
-			}
-
-			// Grab out ItemID, Item IconID, and ItemName
-			int itemID = GetIntFromString(itemFilterStrSplit.at(0), -1);
-			int itemIconID = GetIntFromString(itemFilterStrSplit.at(1), -1);
-			std::string itemName = itemFilterStrSplit.at(2);
-
-			// If valid (actual numbers and some sort of text, set filter with selected types
-			if (itemID != -1 && itemIconID != -1 && !itemName.empty())
-			{
-				pLootFiltersManager->SetItemLootFilter(itemID, itemIconID, itemName.data(), filterTypes);
-				WriteChatf("Item Filter Set for \"%s\"", itemName);
-			}
-			else
-			{
-				WriteChatf("Invalid Item Filter Argument. Check Proper Item ID, Item Icon ID, and Item Name Entered.");
-				return;
-			}
-		}
-	}
-}
+		// Grab Item Icon ID
+		GetArg(szArg, szRest, 1);
+		szRest = GetNextArg(szRest, 1);
 
-// ***************************************************************************
-// Function:    RemoveLootFilterCmd
-// Description: '/removelootfilter' command
-//              Adds the ability to remove advanced loot filters via command line
-// Usage: /removelootfilter itemID
-// ***************************************************************************
-void RemoveLootFilterCmd(SPAWNINFO* pChar, char* szLine)
-{
-	if (pLootFiltersManager)
-	{
-		if (!szLine[0])
+		// Grab/Check Int of Item Icon ID
+		int itemIconID = GetIntFromString(szArg, -1);
+		if (itemIconID == -1)
 		{
-			WriteChatf("Usage: /removelootfilter itemID");
-			WriteChatf(R"(Example: /removelootfilter 13072)");
-			WriteChatf(R"(		Removes AdvLoot Filter for Rat Ears)");
+			WriteChatf("Unable to set AdvLoot Filter, Item Icon ID Invalid.");
 			return;
 		}
 
-		// Grab ItemID from Arguments
-		char itemIDArg[MAX_STRING] = { 0 };
-		GetArg(itemIDArg, szLine, 1);
-		int itemID = GetIntFromString(itemIDArg, -1);
+		// Grab Item Name
+		GetMaybeQuotedArg(szArg, MAX_STRING, szRest, 1);
 
-		// Proper ItemID Found, Remove Filter
-		if (itemID != -1)
+		// Grab/Check String of Item Name
+		std::string itemName(szArg);
+		if (itemName.empty())
 		{
-			pLootFiltersManager->RemoveItemLootFilter(itemID);
-		}
-		else
-		{
-			WriteChatf("Invalid ItemID Entered.");
+			WriteChatf("Unable to set AdvLoot Filter, Item Name Invalid.");
 			return;
 		}
+
+		// Call client function to set advloot filter
+		pLootFiltersManager->SetItemLootFilter(itemID, itemIconID, itemName.data(), filterTypes);
```

Comment:
don't use .data() here, there's no guarantee that it's null terminated, use .c_str() -- but why are you even using a string here because it's directly just `szArg`? Simply use `szArg` and make sure it's not empty (`szArg[0] == 0` should suffice since that's the only validation you are doing anyway)

Reply:
I am used to using QString out of Qt for string manipulation and such, so std::string is close. Mostly just a holdover from when I was initially implementing this.  Should be swapped over to using szArg with that empty check, and the description string now using .c_str().

Reply:
The issue here is the you're doing an extra allocation and copy when you create the string, only to just convert it back into a c-string

Reply:
Yeah I get that, like I said just a holdover from when I was messing around with initial implementation.

Latest push should fix everything thats been mentioned.